### ISSUE TITLE
Failed to serve on production: Uncaught ReferenceError: require is not defined

### ISSUE THREAD

**Issue by chnejohnson** [chnejohnson/vue-dapp#20]
Because Vite is only for ES module, I used [resolve.alias](https://vitejs.dev/config/#resolve-alias) to solve the problem that Vite can't import the built-in node.js package from the third-party modules. Most of the effort is to resolve @walletconnect/web3-provider. There're lots of node.js built-in packages being used by this package.

See these lines:
- https://github.com/chnejohnson/vue-dapp/blob/4b36617ce606258e6c58fab030d558447e706a5c/demo/index.html#L12
- https://github.com/chnejohnson/vue-dapp/blob/4b36617ce606258e6c58fab030d558447e706a5c/vite.config.ts#L12
- https://github.com/chnejohnson/vue-dapp/blob/4b36617ce606258e6c58fab030d558447e706a5c/package.json#L57

Refer to dgrants: https://github.com/dcgtc/dgrants/blob/main/app/vite.config.ts

But in this scenario, when I build the demo files and run `yarn serve:demo`, it failed to run the app and give me: `require is not defined`.

This is hard to resolve. I don't know why `yarn dev` works but it failed on production.

Use branch v0.5.0 to reproduce this issue.

**Comment by chnejohnson** [comment]
The solution of this issue is using `build.rollupOptions.plugins` and `build.commonjsOptions.transformMixedEsModules` in the vite.config.ts, see below:

```ts
// vite.config.ts
import vue from '@vitejs/plugin-vue'
import { resolve } from 'path'
import { defineConfig } from 'vite'
import WindiCSS from 'vite-plugin-windicss'
import inject from '@rollup/plugin-inject'

export default defineConfig({
  root: 'demo/',
  plugins: [vue(), WindiCSS()],
  resolve: {
    alias: {
      'vue-dapp': resolve(__dirname, './src/index.ts'),
    },
  },
  build: {
    rollupOptions: {
      plugins: [inject({ Buffer: ['buffer', 'Buffer'] })],
    },
    commonjsOptions: {
      transformMixedEsModules: true,
    },
  },
})

```

It comes from following two discussions:
- https://github.com/vitejs/vite/discussions/2785#discussioncomment-1450965
- https://github.com/originjs/vite-plugins/issues/9#issuecomment-924713067

**Comment by chnejohnson** [comment]
New solution implemented in PR #98 and [vue3-dapp-starter](https://github.com/chnejohnson/vue3-dapp-starter/blob/main/vite.config.ts)
